### PR TITLE
feat: All Pylons tokens should instead display as USD

### PR DESCRIPTION
**Pull request by AhsanRns**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by MikeSofaer** [comment]
Shouldn't there be an i18n file involved here?

Also as I said in Discord, please output "free" when the currency value is 0 instead of "0 xxxxx"

**Comment by AhsanRns** [comment]
no i18 file has no involvement

these variable are mostly from the constants defined in the app

i will update PR for "free" text

**Comment by AhsanRns** [comment]
PR update for free text

File: easel/lib/utils/constants.dart [hunk 1]
```diff
@@ -134,8 +134,8 @@ const kAgoricSymbol = 'urun';
 const kJunoSymbol = 'ujunox';
 const String kEthereumSymbol = "weth-wei";
 
-const kPylonText = 'Pylon';
-const kUSDText = 'USD';
+const kPylonText = 'IAP';
```
Comment (kjawadDeveloper2) [inline easel/lib/utils/constants.dart:137]: It doesn't match with the variable name. Create a separate variable for it.

File: wallet/lib/pages/detailed_asset_view/owner_view.dart [hunk 2]
```diff
@@ -286,6 +287,11 @@ class _CollapsedBottomMenuState extends State<_CollapsedBottomMenu> {
     final viewModel = context.read<OwnerViewViewModel>();
     final ibcEnumCoins = viewModel.nft.ibcCoins;
 
+    /// this change will reflect only for upylon ibcCoins
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/detailed_asset_view/owner_view.dart:290]:
```suggestion
    /// This change will reflect only for upylon ibcCoins
```

File: wallet/lib/pages/detailed_asset_view/owner_view.dart [hunk 3]
```diff
@@ -286,6 +287,11 @@ class _CollapsedBottomMenuState extends State<_CollapsedBottomMenu> {
...
+    final updatedText =ibcEnumCoins.getAbbrev() == constants.kPYLN_ABBREVATION
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/detailed_asset_view/owner_view.dart:291]:
```suggestion
    final coinWithDenom = ibcEnumCoins.getAbbrev() == constants.kPYLN_ABBREVATION
```

File: wallet/lib/pages/home/wallet_screen/widgets/latest_transactions.dart [hunk 4]
```diff
@@ -72,7 +73,7 @@ class LatestTransactions extends StatelessWidget {
         crossAxisAlignment: CrossAxisAlignment.end,
         children: [
           Text(
-            "${getPrefix(txHistory) ? "" : "+"}${defaultCurrency.convertFromU(txHistory)} ${denomAbbr[defaultCurrency]}",
+            "${getPrefix(txHistory) ? "" : "+"}${denomAbbr[defaultCurrency] == kPYLN_ABBREVATION?"\$":""}${denomAbbr[defaultCurrency] == kPYLN_ABBREVATION?IBCCoins.upylon.pylnToCredit(defaultCurrency.convertFromU(txHistory)):defaultCurrency.convertFromU(txHistory)} ${denomAbbr[defaultCurrency]}",
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/home/wallet_screen/widgets/latest_transactions.dart:76]: Very difficult to understand make a variable out of it.

**Review by kjawadDeveloper2 — APPROVED** [review]
Approving provided that mentioned changes are fixed.

File: easel/lib/utils/constants.dart (hunk 1 above)
Comment (AhsanRns) [reply easel/lib/utils/constants.dart]: done

File: wallet/lib/pages/detailed_asset_view/owner_view.dart (hunk 2 above)
Comment (AhsanRns) [reply wallet/lib/pages/detailed_asset_view/owner_view.dart]: done

File: wallet/lib/pages/detailed_asset_view/owner_view.dart (hunk 3 above)
Comment (AhsanRns) [reply wallet/lib/pages/detailed_asset_view/owner_view.dart]: done

File: wallet/lib/pages/home/wallet_screen/widgets/latest_transactions.dart (hunk 4 above)
Comment (AhsanRns) [reply wallet/lib/pages/home/wallet_screen/widgets/latest_transactions.dart]: done

**Review by MikeSofaer — APPROVED** [review]

**Pull request merged**